src/redux/sagas/motions/expenditures/createStreamingPaymentMotion.ts

```diff
@@ -0,0 +1,185 @@
+    const colonyClient = yield colonyManager.getClient(
+      ClientType.ColonyClient,
+      colonyAddress,
+    );
+    const streamingPaymentsClient: AnyStreamingPaymentsClient = yield call(
+      [colonyManager, colonyManager.getClient],
+      ClientType.StreamingPaymentsClient,
+      colonyAddress,
+    );
+
+    // Get permissions proof of the caller's Funding permission
+    const [permissionDomainId, childSkillIndex] = yield getPermissionProofs(
+      colonyClient,
+      createdInDomain.nativeId,
+      ColonyRole.Arbitration,
+    );
+
+    const encodedCreateStreamingPaymentAction =
+      yield streamingPaymentsClient.interface.encodeFunctionData('create', [
+        permissionDomainId,
+        childSkillIndex,
+        permissionDomainId,
+        childSkillIndex,
+        createdInDomain.nativeId,
+        startTime,
+        endTime ?? TIMESTAMP_IN_FUTURE,
+        interval,
+        recipientAddress,
+        [tokenAddress],
+        [amount],
+      ]);
+
+    const { skillId } = yield call(
+      [colonyClient, colonyClient.getDomain],
+      Id.RootDomain,
+    );
+
+    const { key, value, branchMask, siblings } = yield call(
+      [colonyClient, colonyClient.getReputation],
+      skillId,
+      ADDRESS_ZERO,
+    );
+
+    yield createGroupTransaction(createMotion, batchId, meta, {
+      context: ClientType.VotingReputationClient,
+      methodName: 'createMotion',
+      identifier: colonyAddress,
+      params: [
+        createdInDomain.nativeId,
+        childSkillIndex,
+        streamingPaymentsClient.address,
+        encodedCreateStreamingPaymentAction,
+        key,
+        value,
+        branchMask,
+        siblings,
+      ],
+      group: {
+        title: { id: 'transaction.group.createMotion.title' },
+        description: {
+          id: 'transaction.group.createMotion.description',
+        },
+      },
+    });
+
+    yield initiateTransaction({ id: createMotion.id });
+
+    const {
+      payload: { hash: txHash },
+    } = yield takeFrom(
+      createMotion.channel,
+      ActionTypes.TRANSACTION_HASH_RECEIVED,
+    );
+
+    setTxHash?.(txHash);
+
+    const { type } = yield call(waitForTxResult, createMotion.channel);
+
+    yield apolloClient.mutate<
+      CreateStreamingPaymentMetadataMutation,
+      CreateStreamingPaymentMetadataMutationVariables
+    >({
+      mutation: CreateStreamingPaymentMetadataDocument,
+      variables: {
+        input: {
+          id: getPendingMetadataDatabaseId(colonyAddress, txHash),
+          endCondition,
+          limitAmount,
+        },
+      },
+    });
+
+    if (type === ActionTypes.TRANSACTION_SUCCEEDED) {
+      yield put<Action<ActionTypes.MOTION_STREAMING_PAYMENT_CREATE_SUCCESS>>({
+        type: ActionTypes.MOTION_STREAMING_PAYMENT_CREATE_SUCCESS,
+        meta,
+      });
+
+      window.history.replaceState(
+        {},
+        '',
+        `${window.location.origin}${window.location.pathname}?tx=${txHash}`,
+      );
+    }
```

Comment:
This doesn't seem to work for me, not getting redirected anywhere

Reply:
I think the idea of this pattern is for us to easily access the tx hash once we submit a motion, and use it to manually go to the motion page instead of being redirected which always results in a 404, is this right @willm30?